hypertrace-view-generator/hypertrace-view-generator/src/main/java/org/hypertrace/viewgenerator/generators/SpanEventViewGenerator.java

```diff
@@ -303,4 +310,60 @@ String getRequestUrl(Event event, Protocol protocol) {
     }
     return null;
   }
+
+  /**
+   * For every event compute EventInfo
+   */
+  Map<ByteBuffer, EventInfo> buildEventInfoMap(
+      final Map<ByteBuffer, List<ByteBuffer>> parentToChildrenEventIds,
+      final Map<ByteBuffer, ByteBuffer> childToParentId,
+      final Map<ByteBuffer, Event> eventMap) {
+    Set<ByteBuffer> rootEventIds = new HashSet<>(eventMap.keySet());
+    rootEventIds.removeAll(childToParentId.keySet());
+    Map<ByteBuffer, EventInfo> eventInfoMap = Maps.newHashMap();
+    rootEventIds.forEach(v -> buildEventInfoMapRecursively(
+        parentToChildrenEventIds, eventMap,
+        eventInfoMap, v, null));
+    return eventInfoMap;
+  }
+
+  EventInfo buildEventInfoMapRecursively(
+      final Map<ByteBuffer, List<ByteBuffer>> parentToChildrenEventIds,
+      final Map<ByteBuffer, Event> eventMap,
+      Map<ByteBuffer, EventInfo> eventInfoMap,
+      ByteBuffer eventId, ByteBuffer lastApiTraceId) {
+    ByteBuffer apiTraceId = EnrichedSpanUtils.isEntryApiBoundary(eventMap.get(eventId))
+        ? eventId : lastApiTraceId;
+
+    int totalEventCount = 1;
+    List<ByteBuffer> childrenIds = parentToChildrenEventIds.get(eventId);
+    if (null != childrenIds) {
+      for (ByteBuffer childEventId : childrenIds) {
+        totalEventCount += buildEventInfoMapRecursively(
+            parentToChildrenEventIds, eventMap,
+            eventInfoMap, childEventId, apiTraceId).getTotalEventCount();
+      }
+    }
+    EventInfo eventInfo = new EventInfo(totalEventCount, apiTraceId);
+    eventInfoMap.put(eventId, eventInfo);
+    return eventInfo;
+  }
+
+  private static class EventInfo {
+    private final int totalEventCount;
+    private final ByteBuffer apiTraceId;
+
+    public EventInfo(int totalEventCount, ByteBuffer apiTraceId) {
+      this.totalEventCount = totalEventCount;
+      this.apiTraceId = apiTraceId;
```

Comment:
Is it possible for an event to belong to multiple api traces (not actually sure how we define them)?

Reply:
Yes, but we set the ApiTraceId for the event based on the lowest ancestor which is Api Entry Span.

This was being computed using this method https://github.com/hypertrace/hypertrace-ingester/blob/main/hypertrace-trace-enricher/enriched-span-constants/src/main/java/org/hypertrace/traceenricher/enrichedspan/constants/utils/EnrichedSpanUtils.java#L216
But if we are iterating the tree depth first, it makes sense to compute ApiTraceId on the way.

Reply:
So take a structure like: A (entry) -> B (exit) -> C (entry)
My understanding is that:
- Spans A + B would have the same `apiTraceId`
- Span C would have a different `apiTraceId`
- The total event counts for each span would be `a -> 3` , `b -> 2` and `c -> 1`

Is that accurate?

If yes, I think this can lead to a lot of confusing scenarios. For example, an api trace entry showing a large span count in the hundreds, but when drilled into it, only a couple of spans show up in the trace visualization. On the other hand, if we only show the span count of the spans that belong to that api trace, there's important information missing. These are some of the issues I was attempting to allude to on the issue discussion, but was probably unclear. Implementation aside, I'm not sure it's clear what we actually want to show here to have a consistent UX.

Reply:
> Is that accurate?

Yes 

> If yes, I think this can lead to a lot of confusing scenarios. For example, an api trace entry showing a large span count in the hundreds, but when drilled into it, only a couple of spans show up in the trace visualization.

I agree. But the idea was to keep this field similar to `duration` that we show.